crates/topos-tce-storage/src/lib.rs

```diff
@@ -1,11 +1,88 @@
+//!</picture>
+//!
+//! ### Definitions and Responsibilities
+//!
+//! As illustrated above, multiple `stores` are exposed in the library using various `tables`.
+//!
+//! The difference between a `store` and a `table` is that the `table` is responsible for storing
+//! the data while the `store` is responsible for managing the data and its access and behaviour.
+//!
+//! Here's the list of the different stores and their responsibilities:
+//!
+//! - The [`EpochValidatorsStore`](struct@epoch::EpochValidatorsStore) is responsible for managing the list of validators for each `epoch`.
+//! - The [`FullNodeStore`](struct@fullnode::FullNodeStore) is responsible for managing all the persistent data such as [`Certificate`] delivered and associated `streams`.
+//! - The [`IndexStore`](struct@index::IndexStore) is responsible for managing indexes in order to collect information about the broadcast and the network.
+//! - The [`ValidatorStore`](struct@validator::ValidatorStore) is responsible for managing the pending certificates pool and all the transient and volatile data.
+//!
+//! For more information about a `store`, see the related doc.
+//!
+//! Next, we've the list of the different tables and their responsibilities:
+//!
+//! - The [`EpochValidatorsTables`](struct@epoch::EpochValidatorsTables) is responsible for storing the list of validators for each `epoch`.
+//! - The [`ValidatorPerpetualTables`](struct@validator::ValidatorPerpetualTables) is responsible for storing the [`Certificate`] delivered and all the persitent data related to the broadcast.
+//! - The [`ValidatorPendingTables`](struct@validator::ValidatorPendingTables) is responsible for storing the pending certificates pool and all the transient and volatile data.
+//! - The [`IndexTables`](struct@index::IndexTables) is responsible for storing indexes about the delivery of [`Certificate`] such as `target subnet stream`.
+//!
+//! ## Special Considerations
+//!
+//! When using the storage layer, you need to be aware of the following:
+//! - The storage layer is using [rocksdb](https://rocksdb.org/) as a backend, which means that this storage doesn't need external service, as `rocksdb` is embeddable kv store.
+//! - The storage layer is using [`Arc`](struct@std::sync::Arc) to share the stores between threads. It also means that a `store` is only instantiated once.
+//! - Some functions are batching multiple writes in one transaction. But not all functions are using it.
+//!
+//! ## Design Philosophy
+//!
+//! The choice of using [rocksdb](https://rocksdb.org/) as a backend was made because it is a well known and battle tested database.
+//! It is also very fast and efficient when it comes to write and read data.
+//!
+//! Multiple `stores` and `tables` exists in order to allow admin to deal with backups or
+//! snapshots as they see fit. You can pick and choose which `tables` you want to backup without having to backup the whole database.
```

Comment:
There are multiple options to back up database or folder. It depends on which kind of backup the admin want to rely.

If an admin uses volume snapshot or cold backups provided by its system storage engine, there are no issues. If the admin wants to have incremental backups or doesn't have access to storage engine agnostic methods it can be more difficult, we could provide options for admin to define their backup mechanism allowing the storage to execute `RocksDB` backup functions.

Reply:
>If an admin uses volume snapshot or cold backups provided by its system storage engine, there are no issues.

Sure, but the point in the text was that having multiple tables (i.e. Column Families) would somehow make admins life easier and I'm not sure it matters much to them. I think RocksDB backups is a somewhat complex topic and perhaps not super relevant to the code these docs is describing.

Reply:
I removed the last sentences, let me know if it's OK